code/game/atoms.dm

```diff
@@ -436,8 +436,70 @@ var/list/blood_splatter_icons = list()
 	dir = newdir
 
 /atom/on_varedit(modified_var)
+	..()
 	if(!Debug2)
 		admin_spawned = TRUE
+	switch(modified_var)
+		if("color")
+			add_atom_colour(color, ADMIN_COLOUR_PRIORITY)
 
 /atom/proc/mech_melee_attack(obj/mecha/M)
-	return
+	return
+
+
+
+/*
+	Atom Colour Priority System
+	A System that gives finer control over which atom colour to colour the atom with.
+	The "highest priority" one is always displayed as opposed to the default of
+	"whichever was set last is displayed"
+*/
+
+/atom
+	var/list/atom_colours	 //used to store the different colors on an atom
+							//its inherent color, the colored paint applied on it, special color effect etc...
+
+/*
+	Adds an instance of colour_type to the atom's atom_colours list
+*/
+/atom/proc/add_atom_colour(coloration, colour_priority)
+	if(!atom_colours)
+		atom_colours = list("", "", "", "") //four priority levels currently.
+	if(!coloration)
+		return
+	if(colour_priority > atom_colours.len)
+		return
+	atom_colours[colour_priority] = coloration
+	update_atom_colour()
+
+
+/*
+	Removes an instance of colour_type from the atom's atom_colours list
+*/
+/atom/proc/remove_atom_colour(colour_priority)
+	if(!atom_colours)
+		atom_colours = list("", "", "", "")
+	if(colour_priority > atom_colours.len)
+		return
+	atom_colours[colour_priority] = ""
+	update_atom_colour()
+
+
+/*
+	Resets the atom's color to null, and then sets it to the highest priority
+	colour available
+*/
+/atom/proc/update_atom_colour()
+	if(!atom_colours)
+		atom_colours = list("", "", "", "")

```

Comment:
either:
Add a define for the number of priorities (eg: ```#define COLOUR_PRIORITY_AMOUNT 4```) and then do 
```
atom_colours = list()
atom_colours.len = COLOUR_PRIORITY_AMOUNT
```

or add a global list of ```list("", "", "", "")``` to ```.Copy()``` from in resets, as it stands you use the list ```list("","","","")``` in atleast 3 places, and that's 3 places it's going to break/need updating if another priority is added.